kedro/framework/cli/starters.py

```diff
@@ -62,6 +142,69 @@ def _remove_readonly(func: Callable, path: Path, excinfo: Tuple):  # pragma: no
     func(path)
 
 
+def _is_valid_starter_entrypoint_config(
+    module_name: str, config: KedroStarterSpec
+) -> bool:
+    if not isinstance(config, KedroStarterSpec):
+        warn(
+            f"The starter configuration loaded from module {module_name}"
+            f"should be a 'KedroStarterSpec', got '{type(config)}' instead"
+        )
+        return False
+    return True
+
+
+def _get_starters_aliases() -> List[KedroStarterSpec]:
+    """This functions lists all the starters aliases declared in
+    the core repo and in plugins entry points.
+    The output looks like:
+    [
+        KedroStarterSpec(
+            name="astro-airflow-iris",
+            template_path="git+https://github.com/kedro-org/kedro-starters.git",
+            directory=None,
+            origin="kedro",
+        ),
+        KedroStarterSpec(
+            name="astro-iris",
+            template_path="git+https://github.com/kedro-org/kedro-starters.git",
+            directory="astro-airflow-iris",
+            origin="kedro",
+        ),
+    ]
+    """
+    # add an extra key to indicate from where the plugin come from
+    starters_aliases = _OFFICIAL_STARTERS_SPECS
+
+    existing_names: Dict[str, str] = {}  # dict {name: module_name}
+    for starter_entry_point in importlib_metadata.entry_points().select(
+        group=ENTRY_POINT_GROUPS["starters"]
+    ):
+        module_name = starter_entry_point.module.split(".")[0]
+        for starter_config in starter_entry_point.load():
+            is_valid_config_status = _is_valid_starter_entrypoint_config(
+                module_name, starter_config
+            )
+            if not is_valid_config_status:
+                click.secho(
+                    f"Starter alias `{starter_config.name}` from `{module_name}` "
+                    f"has been ignored as the config is invalid and cannot be loaded",
+                    fg="yellow",
+                )
```

Comment:
Doesn't this just repeat the warning from inside `_is_valid_starter_entrypoint_config`? Could you just do the check inline here?
```
if not isinstance(config, KedroStarterSpec):
    click.secho("...")
elif ...
```
and then remove `_is_valid_starter_entrypoint_config` altogether?

Reply:
I think so, this is pretty much a leftover method that doesn't do much now since all the validation logic is removed.